db/revision.go

```diff
@@ -425,31 +425,49 @@ func compareRevIDs(id1, id2 string) int {
 	return 0
 }
 
-func stripSpecialProperties(body Body) Body {
-	stripped := Body{}
-	for key, value := range body {
-		if key == "" || key[0] != '_' || key == BodyAttachments || key == BodyDeleted {
-			stripped[key] = value
+// stripSpecialProperties returns a copy of the given body with all underscore-prefixed keys removed, except _attachments and _deleted.
+func stripSpecialProperties(b Body) Body {
+	return stripSpecialPropertiesExcept(b, []string{BodyAttachments, BodyDeleted})
+}
+
+// stripAllSpecialProperties returns a copy of the given body with all underscore-prefixed keys removed.
+func stripAllSpecialProperties(b Body) Body {
+	return stripSpecialPropertiesExcept(b, nil)
+}
+
+// stripSpecialPropertiesExcept returns a copy of the given body with all underscore-prefixed keys removed, except those given.
+func stripSpecialPropertiesExcept(b Body, exceptions []string) Body {
+	// Assume no properties removed for the initial capacity to reduce allocs on large docs.
+	stripped := make(Body, len(b))
+	for k, v := range b {
+		if k == "" || k[0] != '_' ||
+			base.StringSliceContains(exceptions, k) {
```

Comment:
Is there non-zero overhead associated with calling StringSliceContains for each key when exceptions is empty?

Reply:
Negligable difference. Updated benchmark to include `stripAllSpecialProperties`

```
BenchmarkStripSpecialProperties/no_specialAllSpecial-12         	 1289540	       925 ns/op	    1202 B/op	       2 allocs/op
BenchmarkStripSpecialProperties/no_specialAllSpecial-12         	 1293424	       929 ns/op	    1202 B/op	       2 allocs/op
BenchmarkStripSpecialProperties/no_specialAllSpecial-12         	 1282030	       926 ns/op	    1202 B/op	       2 allocs/op

BenchmarkStripSpecialProperties/no_specialOldSpecialSpecial-12  	 1291028	       927 ns/op	    1202 B/op	       2 allocs/op
BenchmarkStripSpecialProperties/no_specialOldSpecialSpecial-12  	 1293970	       930 ns/op	    1202 B/op	       2 allocs/op
BenchmarkStripSpecialProperties/no_specialOldSpecialSpecial-12  	 1267436	       932 ns/op	    1202 B/op	       2 allocs/op

BenchmarkStripSpecialProperties/specialAllSpecial-12            	 1301296	       924 ns/op	    1202 B/op	       2 allocs/op
BenchmarkStripSpecialProperties/specialAllSpecial-12            	 1303908	       921 ns/op	    1202 B/op	       2 allocs/op
BenchmarkStripSpecialProperties/specialAllSpecial-12            	 1299585	       922 ns/op	    1202 B/op	       2 allocs/op

BenchmarkStripSpecialProperties/specialOldSpecialSpecial-12     	 1305183	       916 ns/op	    1202 B/op	       2 allocs/op
BenchmarkStripSpecialProperties/specialOldSpecialSpecial-12     	 1309207	       915 ns/op	    1202 B/op	       2 allocs/op
BenchmarkStripSpecialProperties/specialOldSpecialSpecial-12     	 1307422	       920 ns/op	    1202 B/op	       2 allocs/op
```